### PR TITLE
Add jest and supertest, build first couple test cases

### PR DESCRIPTION
**Pull request by TheMallen**
This PR adds jest and supertest as dependencies and adds the first couple tests to ShopifyAuth :)

### CODE REVIEW

File: routes/shopifyAuth.js [hunk 1]
```diff
@@ -14,24 +14,26 @@ module.exports = function createShopifyAuthRouter({
 
   // This function initializes the Shopify OAuth Process
   router.get('/', function(request, response) {
-    const { query, session } = request;
+    const { query } = request;
```
Comment (TheMallen) [inline routes/shopifyAuth.js]: it didn't make sense to do half the session in this one and half in the other

File: routes/shopifyAuth.js [hunk 2]
```diff
@@ -83,6 +85,7 @@ module.exports = function createShopifyAuthRouter({
           }
 
           request.session.accessToken = accessToken;
+          request.session.shop = shop;
```
Comment (TheMallen) [inline routes/shopifyAuth.js]: this will error without having `express-session` hooked up, opened #21 about this

File: routes/shopifyAuth.js [hunk 3]
```diff
@@ -14,24 +14,26 @@ module.exports = function createShopifyAuthRouter({
...
-        `<!DOCTYPE html>
-        <html>
-          <head>
-            <script type="text/javascript">
-              window.top.location.href = "${redirectTo}${redirectParams}"
-            </script>
-          </head>
-        </html>`
-      );
+    if (shop == null) {
```
Comment (TheMallen) [inline routes/shopifyAuth.js]: before we would just do nothing and let the request timeout if there was no shop

File: package.json [hunk 4]
```diff
@@ -19,18 +19,21 @@
   },
   "scripts": {
     "pretty": "prettier --single-quote --trailing-comma es5 --write {middleware,routes,shopStore}{/*,/**/*}.js",
-    "precommit": "yarn run pretty"
+    "precommit": "yarn run pretty",
```
Comment (AWaselnuk) [inline package.json]: This is unrelated but I'm pretty sure this will never work without [Husky](https://github.com/typicode/husky). Issue opened: https://github.com/Shopify/shopify-express/issues/22

File: routes/tests/shopifyAuth.test.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+  describe('/', () => {
+    it('responds to get requests by returning a redirect page', async () => {
+      const {app, server} = await createServer();
+      const {status, text} = await request(app).get('/?shop=shop1').then((resp) => resp);
+
+      expect(status).toBe(200);
+      expect(text).toMatchSnapshot();
+
+      server.close();
```
Comment (AWaselnuk) [inline routes/tests/shopifyAuth.test.js]: I think we might want to remove the locally scoped `server` given the `afterEach` block.

File: routes/tests/shopifyAuth.test.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+let server;
+describe('shopifyAuth', async () => {
+  afterEach(() => {
+    server.close();
+  });
+
+  describe('/', () => {
+    it('responds to get requests by returning a redirect page', async () => {
+      const {app, server} = await createServer();
+      const {status, text} = await request(app).get('/?shop=shop1').then((resp) => resp);
```
Comment (AWaselnuk) [inline routes/tests/shopifyAuth.test.js]: Would it be clearer to pass `() => {}` to `then`?

File: routes/tests/shopifyAuth.test.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+  app.use('/', createShopifyAuthRouter({
+    apiKey: 'key',
+    secret: 'secret',
+    scope: ['scope'],
+    shopStore: new MemoryStrategy(),
+    afterAuth,
+  }));
+
+  server = http.createServer(app);
```
Comment (AWaselnuk) [inline routes/tests/shopifyAuth.test.js]: I still find this variable a _little_ hard to follow but I do like this `createServer` function.

File: routes/tests/shopifyAuth.test.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+      expect(status).toBe(400);
+      expect(text).toMatchSnapshot();
+
+      server.close();
+    });
+  });
+})
+
+function createServer({afterAuth = jest.fn()} = {}) {
```
Comment (AWaselnuk) [inline routes/tests/shopifyAuth.test.js]: Why choose `jest.fn()` here? Would that also be an option for our thing to invoke `then` for `supertest`?

File: routes/tests/shopifyAuth.test.js (hunk 6 above)
Comment (TheMallen) [reply routes/tests/shopifyAuth.test.js]: Have to return the wrapped value or else the await won't give us what we want. :/

File: routes/tests/shopifyAuth.test.js (hunk 7 above)
Comment (TheMallen) [reply routes/tests/shopifyAuth.test.js]: Mhm, I'd love a better solution.

File: routes/tests/shopifyAuth.test.js (hunk 8 above)
Comment (TheMallen) [reply routes/tests/shopifyAuth.test.js]: `jest.fn()` is a jest mock, so we're returning it here so we can do things like test that it was called with what we expect at the end of the function.

File: package.json (hunk 4 above)
Comment (TheMallen) [reply package.json]: right good catch
